Example/AblyChatExample/Mocks/Mocks.swift

```diff
@@ -0,0 +1,234 @@
+import Ably
+import AblyChat
+
+actor MockChatClient: ChatClient {
+    let realtime: RealtimeClient
+    nonisolated let clientOptions: ClientOptions
+    nonisolated let rooms: Rooms
+
+    init(realtime: RealtimeClient, clientOptions: ClientOptions?) {
+        self.realtime = realtime
+        self.clientOptions = clientOptions ?? .init()
+        self.rooms = MockRooms(clientOptions: self.clientOptions)
+    }
+    
+    nonisolated var connection: any Connection {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var clientID: String {
+        fatalError("Not yet implemented")
+    }
+}
+
+actor MockRooms: Rooms {
+    let clientOptions: ClientOptions
+    private var rooms = [String: MockRoom]()
+    
+    func get(roomID: String, options: RoomOptions) async throws -> any Room {
+        if let room = rooms[roomID] {
+            return room
+        }
+        let room = MockRoom(roomID: roomID, options: options)
+        rooms[roomID] = room
+        return room
+    }
+    
+    func release(roomID: String) async throws {
+        fatalError("Not yet implemented")
+    }
+    
+    init(clientOptions: ClientOptions) {
+        self.clientOptions = clientOptions
+    }
+}
+
+actor MockRoom: Room {
+    internal nonisolated let roomID: String
+    internal nonisolated let options: RoomOptions
+    
+    internal init(roomID: String, options: RoomOptions) {
+        self.roomID = roomID
+        self.options = options
+    }
+    
+    lazy nonisolated var messages: any Messages = MockMessages(clientID: "AblyTest", roomID: roomID)
+
+    lazy nonisolated var presence: any Presence = MockPresence(clientID: "AblyTest", roomID: roomID)
+
+    lazy nonisolated var reactions: any RoomReactions = MockRoomReactions(clientID: "AblyTest", roomID: roomID)
+
+    lazy nonisolated var typing: any Typing = MockTyping(clientID: "AblyTest", roomID: roomID)
+
+    nonisolated var occupancy: any Occupancy {
+        fatalError("Not yet implemented")
+    }
+
+    nonisolated var status: any RoomStatus {
+        fatalError("Not yet implemented")
+    }
+
+    func attach() async throws {
+        fatalError("Not yet implemented")
+    }
+
+    func detach() async throws {
+        fatalError("Not yet implemented")
+    }
+}
+
+actor MockMessages: Messages {
+    let clientID: String
+    let roomID: String
+    let channel: RealtimeChannel
+    
+    private var mockSubscription: MockMessageSubscription
```

Comment:
You can't reuse the same `MockMessageSubscription`. It wraps an `AsyncStream`, which is only designed to be iterated over once (I think that the terminology I’ve seen used is that it’s "unicast"). This means that a user who calls `messages.subscribe(bufferingPolicy:)` twice and then iterates over the returned sequences will see that each iteration does not receive all of the messages. (To see this in a more concrete fashion, change your `AblyChatExampleApp` to include two `ContentView` instances instead of one, and you’ll see that when you send a message it only appears in one of the content views.

I’m not yet sure what the best way to solve this is — my first thought was to say to just create a new `MockMessageSubscription` each time `messages.subscribe(bufferingPolicy:)` is called, but this wouldn't be right because the timers wouldn't be in sync and the random elements would be different. The ideal thing would be to use some sort of "broadcast" operator (see e.g. [this swift-async-algorithms discussion](https://forums.swift.org/t/swift-async-algorithms-proposal-broadcast-previously-shared/61210)) to allow multiple iterations over a single subscription. Will think about it a bit more, keen to hear your thoughts too.

Ditto for the other subscriptions (reactions etc).

Reply:
>include two ContentView instances instead of one

This should not happen in real app 😆

>but this wouldn't be right because the timers wouldn't be in sync and the random elements would be different

But we shouldn't care about random ones too much, right?

>The ideal thing would be to use some sort of "broadcast" operator

If it will not require another change of programming paradigm and can be just "attached" via extension or something I don't see any obstacles in trying this "magic trick".

Reply:
https://github.com/ably-labs/ably-chat-swift/pull/34/commits/2626143fca70472ef284fe8d0d54954347e0c01b

Reply:
Issue https://github.com/ably-labs/ably-chat-swift/issues/44